connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -99,7 +100,7 @@ public Response createConnector(final @QueryParam("forward") Boolean forward,
         herder.putConnectorConfig(name, configs, false, cb);
         Herder.Created<ConnectorInfo> info = completeOrForwardRequest(cb, "/connectors", "POST", createRequest,
                 new TypeReference<ConnectorInfo>() { }, new CreatedConnectorInfoTranslator(), forward);
-        return Response.created(URI.create("/connectors/" + name)).entity(info.result()).build();
+        return Response.created(URI.create("/connectors/" + URLEncoder.encode(name, "UTF-8"))).entity(info.result()).build();
```

Comment:
I am not sure if URLEncoder will behave correctly in this case ([see this SO](https://stackoverflow.com/questions/14321873/java-url-encoding-urlencoder-vs-uri)). The one large difference to the URI object is, that it will encode spaces as + which I think will be interpreted as an actual plus sign by connect (tested very briefly, may need further investigation with your actual changes). The correct encoding would be %20 - since I believe we want to allow spaces in the middle of connector names. I think using the URI object for encoding here would be a better choice.

Reply:
Thanks for the comment, @soenkeliebau. Regarding the '/', there is a check in createConnector which throws a BadRequestException if there is a '/' in the name of the connector. Hence, we should not encounter a scenario where a '/' needs to be encoded in the name. 

Characters like '@' are not percent encoded because they don't have a special meaning (yet) even though they are reserved (the URIBuilders probably skip such characters from being percent encoded to keep the URL short to better support HTTP clients which have lower limits on the length of the URI than others).

Let me know what you think.

Reply:
Also, I am going to switch to use javax.ws.rs.core.UriBuilder to construct the Location URI. It seems better with handling special characters.